src/components/AppliedStateFilter/AppliedStateFilter.tsx

```diff
@@ -1,18 +1,33 @@
-import { useMemo } from 'react';
+import { useCallback, useMemo } from 'react';
+import { useLatest } from '@taskany/bricks';
+import { TagCleanButton } from '@taskany/bricks/harmony';
 
 import { AppliedFilter } from '../AppliedFilter/AppliedFilter';
 import { StateDropdown } from '../StateDropdown/StateDropdown';
 import { trpc } from '../../utils/trpcClient';
+import { StateType } from '../../../trpc/inferredTypes';
 
 interface AppliedStateFilterProps {
     label?: string;
     value?: string[];
     stateTypes?: string[];
     readOnly?: boolean;
+    onChange?: (values?: string[]) => void;
+    onStateTypeChange?: (values?: StateType[]) => void;
+    onClose?: () => void;
 }
 
-export const AppliedStateFilter = ({ label, value, stateTypes, readOnly }: AppliedStateFilterProps) => {
+export const AppliedStateFilter = ({
+    label,
+    value,
+    stateTypes,
+    readOnly,
+    onChange,
+    onStateTypeChange,
```

Comment:
we don't need additional prop `onStateTypeChange`, If we just reset `stateType` on `state` change